### PR TITLE
⏱ [Manifest Cache] Fix for manifests are never cached when using remote caching

### PR DESCRIPTION
**Pull request by danyf90**
Resolves https://github.com/tuist/tuist/issues/3467

### Short description 📝

Rename internal variables to start with `TUIST_CONFIG` and ignore them in manifests caching.
It also reverts https://github.com/tuist/tuist/pull/3337 which was not working anyway as for manifest loading the `manifestLoadingVariables` property is used, which strips the non `TUIST_` env variables before passing it to the `ProjectDescription.Environment`

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/TuistSupport/Constants.swift [hunk 1]
```diff
@@ -60,17 +60,31 @@ public enum Constants {
     /// These variables are not supposed to be used by end users
     /// But only eg. for acceptance tests and other cases needed internally
     public enum EnvironmentVariables {
-        public static let verbose = "TUIST_VERBOSE"
-        public static let colouredOutput = "TUIST_COLOURED_OUTPUT"
-        public static let versionsDirectory = "TUIST_VERSIONS_DIRECTORY"
+        public static let verbose = "TUIST_CONFIG_VERBOSE"
+        public static let colouredOutput = "TUIST_CONFIG_COLOURED_OUTPUT"
+        public static let versionsDirectory = "TUIST_CONFIG_VERSIONS_DIRECTORY"
         @available(*, deprecated, message: "This is used only for acceptance tests, see https://github.com/tuist/tuist/issues/2777")
```
Comment (danyf90) [inline Sources/TuistSupport/Constants.swift:66]: @pepibumur not related to this PR, but if we don't plan to do anything for this, can we remove this deprecation, and maybe made it clear from the name that it shouldn't be used for other purposes? IMO it's a bit annoying to always have that warning popping up knowing we can't fix it

**Review by bolismauro — COMMENTED** [review]
@danyf90 what about having just the `TUIST_CONFIG_` variant in `EnvironmentVariables` and do some pre processing somewhere that adds, for each `TUIST_<something>` the `TUIST_CONFIG_` counterpart? (it should be done just for the keys that have a legacy counterpart)

The problem would be to manage the code that accesses `ProcessInfo.processInfo.environment` directly tho

**Comment by danyf90** [comment]
As you said, I think the main problem would be to manage the code that accesses `ProcessInfo.processInfo.environment` directly.
As 2.0 release should be not too far, I would go for the simpler approach (the one in the PR), and will open another PR on the tuist 2.0 branch to remove the deprecated ones

File: Sources/TuistSupport/Constants.swift (hunk 1 above)
Comment (pepibumur) [reply Sources/TuistSupport/Constants.swift:66]: Agree. Let's remove it.

File: Sources/TuistSupport/Constants.swift [hunk 2]
```diff
@@ -60,17 +60,31 @@ public enum Constants {
...
+        public static let automationPath = "TUIST_CONFIG_AUTOMATION_PATH"
+        public static let queueDirectory = "TUIST_CONFIG_QUEUE_DIRECTORY"
+        public static let cloudToken = "TUIST_CONFIG_CLOUD_TOKEN"
+        public static let cacheManifests = "TUIST_CONFIG_CACHE_MANIFESTS"
+        public static let statsOptOut = "TUIST_CONFIG_STATS_OPT_OUT"
+        public static let githubAPIToken = "TUIST_CONFIG_GITHUB_API_TOKEN"
+        public static let detailedLog = "TUIST_CONFIG_DETAILED_LOG"
+        public static let osLog = "TUIST_CONFIG_OS_LOG"
+
+        public static let deprecatedVerbose = "TUIST_VERBOSE"
```
Comment (pepibumur) [inline Sources/TuistSupport/Constants.swift:77]: Is it ok if we remove those in 2.0?

**Review by pepibumur — APPROVED** [review]

File: Sources/TuistSupport/Constants.swift (hunk 2 above)
Comment (danyf90) [reply Sources/TuistSupport/Constants.swift:77]: yes, once we merge these I'll open a PR there

**Review by fortmarek — APPROVED** [review]

**Pull request merged**